### PR TITLE
PhoneOTPScreen/UI

### PR DESCRIPTION
**Pull request by vaishnavi-agr**
Designed the OTP screen of phone verification
WhatsApp Image 2024-06-20 at 4 40 28 PM

### CODE REVIEW

**Comment by Jha2508** [comment]
@vaishnavi-agr please have this kind of structure : 
- src
    - screens
         - OtpScreen
               - index.tsx
               - styles.ts

File: src/UI/Components/Flipcoin.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import React, {useEffect, useRef} from 'react';
+import {Animated, SafeAreaView, StyleSheet} from 'react-native';
+
+const Flipcoin = () => {
```
Comment (Jha2508) [inline src/UI/Components/Flipcoin.tsx]: documentation

File: src/UI/Components/Flipcoin.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  const animatedValue = useRef(new Animated.Value(0)).current;
+
+  const startFlipAnimation = () => {
```
Comment (Jha2508) [inline src/UI/Components/Flipcoin.tsx]: document this function

File: src/UI/Components/Flipcoin.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        }),
+      ]),
+    ).start();
+  };
+
+  useEffect(() => {
+    startFlipAnimation();
+  }, []);
+
+  const setInterpolate = animatedValue.interpolate({
```
Comment (Jha2508) [inline src/UI/Components/Flipcoin.tsx]: single line comment for this value about what is it's use

File: src/UI/Components/Flipcoin.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    inputRange: [0, 180],
+    outputRange: ['0deg', '180deg'],
+  });
+
+  const rotateYAnimatedStyle = {
```
Comment (Jha2508) [inline src/UI/Components/Flipcoin.tsx]: same as above

File: src/UI/Components/Flipcoin.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    <SafeAreaView style={{flex: 1}}>
+      <Animated.Image
+        source={require('../../Assets/logo.png')}
+        style={[rotateYAnimatedStyle, styles.imagsp]}
+      />
+    </SafeAreaView>
+  );
+};
+
+const styles = StyleSheet.create({
```
Comment (Jha2508) [inline src/UI/Components/Flipcoin.tsx]: move styles out of it. similar structure for components can be followed as I have told for screens

File: src/UI/Components/OTPinput.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
+// OTPInputs.tsx
+
+import React, {useRef} from 'react';
+import {StyleSheet, TextInput, View} from 'react-native';
+
+interface OTPInputsProps {
```
Comment (Jha2508) [inline src/UI/Components/OTPinput.tsx]:
```suggestion
type OTPInputsProps {
```

File: src/UI/Components/OTPinput.tsx [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  otp: string[];
+  handleOtpChange: (index: number, value: string) => void;
+  isChecked: boolean;
+}
+
+const OTPInputs: React.FC<OTPInputsProps> = ({
```
Comment (Jha2508) [inline src/UI/Components/OTPinput.tsx]: do not use React.FC, let us just have a Arrow function taking otpInputsProps

Comment (Jha2508) [reply src/UI/Components/OTPinput.tsx]: +documentation

File: src/UI/Components/TimerDisplay.tsx [hunk 8]
```diff
@@ -0,0 +1,35 @@
+// TimerDisplay.tsx
+
+import React from 'react';
+import {StyleSheet, Text, View} from 'react-native';
+
+interface TimerDisplayProps {
```
Comment (Jha2508) [inline src/UI/Components/TimerDisplay.tsx]:
```suggestion
type TimerDisplayProps {
```

File: src/UI/Screens/OTPScreens/OTPScreen.tsx [hunk 9]
```diff
@@ -0,0 +1,64 @@
+import {StyleSheet, View, TouchableOpacity, Text} from 'react-native';
+import React, {useState, useEffect} from 'react';
+import {SafeAreaView} from 'react-native-safe-area-context';
+import Flipcoin from '../Components/Flipcoin';
+import OTPInputs from '../Components/OTP_input';
+import TimerDisplay from '../Components/TimerDisplay';
+import styles from './OTP_Screen.style';
+export default function OTP_Screen() {
```
Comment (Jha2508) [inline src/UI/Screens/OTPScreens/OTPScreen.tsx]:
```suggestion
export default function OTPScreen() {
```

Comment (Jha2508) [reply src/UI/Screens/OTPScreens/OTPScreen.tsx]: + documentation

File: src/UI/Components/TimerDisplay.tsx [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+  return (
+    <View>
+      <Text style={styles.timerText}>
+        {formattedMinutes}:{formattedSeconds}
+      </Text>
+    </View>
+  );
+};
+
+const styles = StyleSheet.create({
```
Comment (Jha2508) [inline src/UI/Components/TimerDisplay.tsx]: move styles out

File: src/UI/Components/TimerDisplay.tsx [hunk 11]
```diff
@@ -0,0 +1,35 @@
+// TimerDisplay.tsx
+
+import React from 'react';
+import {StyleSheet, Text, View} from 'react-native';
+
+interface TimerDisplayProps {
+  seconds: number; // Define seconds prop as number type
+}
+
+const TimerDisplay: React.FC<TimerDisplayProps> = ({seconds}) => {
```
Comment (Jha2508) [inline src/UI/Components/TimerDisplay.tsx]: arrow function + docs

File: src/UI/Components/Flipcoin.tsx (hunk 1 above)
Comment (vaishnavi-agr) [reply src/UI/Components/Flipcoin.tsx]: Done

File: src/UI/Components/Flipcoin.tsx (hunk 2 above)
Comment (vaishnavi-agr) [reply src/UI/Components/Flipcoin.tsx]: Done

File: src/UI/Components/Flipcoin.tsx (hunk 3 above)
Comment (vaishnavi-agr) [reply src/UI/Components/Flipcoin.tsx]: Controls the rotation of the coin being animated

File: src/UI/Components/Flipcoin.tsx (hunk 4 above)
Comment (vaishnavi-agr) [reply src/UI/Components/Flipcoin.tsx]: Done

File: src/UI/Components/Flipcoin.tsx (hunk 5 above)
Comment (vaishnavi-agr) [reply src/UI/Components/Flipcoin.tsx]: Done

File: src/UI/Components/OTPinput.tsx (hunk 6 above)
Comment (vaishnavi-agr) [reply src/UI/Components/OTPinput.tsx]: Done

File: src/UI/Components/OTPinput.tsx (hunk 7 above)
Comment (vaishnavi-agr) [reply src/UI/Components/OTPinput.tsx]: Done

File: src/UI/Components/TimerDisplay.tsx (hunk 8 above)
Comment (vaishnavi-agr) [reply src/UI/Components/TimerDisplay.tsx]: Done

File: src/UI/Screens/OTPScreens/OTPScreen.tsx (hunk 9 above)
Comment (vaishnavi-agr) [reply src/UI/Screens/OTPScreens/OTPScreen.tsx]: Done

File: src/UI/Components/TimerDisplay.tsx (hunk 10 above)
Comment (vaishnavi-agr) [reply src/UI/Components/TimerDisplay.tsx]: Done

File: src/UI/Components/TimerDisplay.tsx (hunk 11 above)
Comment (vaishnavi-agr) [reply src/UI/Components/TimerDisplay.tsx]: Done

**Review by Armaggeddon12 — APPROVED** [review]

**Pull request merged**